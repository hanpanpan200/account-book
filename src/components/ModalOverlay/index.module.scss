@import "../../styles/colors";
@import "../../styles/global";

.show {
  visibility: visible;
}
.hide {
  visibility: hidden;
}

.container {
  height: 100vh;
  width: 100vw;
  z-index: $z-index-top;
  background-color: rgba(0, 0, 0, .6);
  position: absolute;
  top: 0;
}