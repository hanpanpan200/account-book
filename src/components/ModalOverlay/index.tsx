import React from 'react';
import classNames from 'classnames/bind';

import styles from './index.module.scss';

const cx = classNames.bind(styles);

interface Props {
  visible: boolean
  children: JSX.Element
}

const ModalOverlay:React.FC<Props> = ({children, visible}) => {
  return (
    <div className={cx(styles.container, {[styles.show]: visible, [styles.hide]: !visible })}>
      {children}
    </div>
  )
}

export default ModalOverlay;