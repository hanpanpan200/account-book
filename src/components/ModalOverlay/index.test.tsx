import React from 'react';
import { render } from '@testing-library/react';
import ModalOverlay from './index';

describe('ModalOverlay', () => {
  it('should render correctly when visible', () => {
    const { container } = render(
      <ModalOverlay visible>
        <div>Modal Content</div>
      </ModalOverlay>
    );
    expect(container).toMatchSnapshot();
  });
  it('should render correctly when invisible', () => {
    const { container } = render(
      <ModalOverlay visible={false}>
        <div>Modal Content</div>
      </ModalOverlay>
    );
    expect(container).toMatchSnapshot();
  });
});


