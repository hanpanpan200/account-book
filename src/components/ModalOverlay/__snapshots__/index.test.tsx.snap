// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalOverlay should render correctly when invisible 1`] = `
<div>
  <div
    class="container hide"
  >
    <div>
      Modal Content
    </div>
  </div>
</div>
`;

exports[`ModalOverlay should render correctly when visible 1`] = `
<div>
  <div
    class="container show"
  >
    <div>
      Modal Content
    </div>
  </div>
</div>
`;
